examples-gallery/plot_two_link_pendulum_on_a_cart.py

```diff
@@ -0,0 +1,276 @@
+# %%
+prob.plot_constraint_violations(solution)
+
+# %%
+# animate the solution.
+P1_x = np.empty(num_nodes)
+P1_y = np.empty(num_nodes)
+P2_x = np.empty(num_nodes)
+P2_y = np.empty(num_nodes)
+P3_x = np.empty(num_nodes)
+P3_y = np.empty(num_nodes)
+
+P1_loc = [me.dot(P1.pos_from(O), uv) for uv in [N.x, N.y]]
+P2_loc = [me.dot(P2.pos_from(O), uv) for uv in [N.x, N.y]]
+P3_loc = [me.dot(P3.pos_from(O), uv) for uv in [N.x, N.y]]
+
+qL = q_ind + u_ind
+pL_vals = list(constant_symbols)
+P1_loc_lam = sm.lambdify(qL + pL_vals, P1_loc, cse=True)
+P2_loc_lam = sm.lambdify(qL + pL_vals, P2_loc, cse=True)
+P3_loc_lam = sm.lambdify(qL + pL_vals, P3_loc, cse=True)
+
+for i in range(num_nodes):
+    q_1 = solution[i]
+    q_2 = solution[i + num_nodes]
+    q_3 = solution[i + 2 * num_nodes]
+    u_1 = solution[i + 3 * num_nodes]
+    u_2 = solution[i + 4 * num_nodes]
+    u_3 = solution[i + 5 * num_nodes]
+    P1_x[i], P1_y[i] = P1_loc_lam(q_1, q_2, q_3, u_1, u_2, u_3, *list(par_map.values()))
+    P2_x[i], P2_y[i] = P2_loc_lam(q_1, q_2, q_3, u_1, u_2, u_3, *list(par_map.values()))
+    P3_x[i], P3_y[i] = P3_loc_lam(q_1, q_2, q_3, u_1, u_2, u_3, *list(par_map.values()))
+    
+
+# needed to give the picture the right size.
+xmin = min(np.min(P1_x),np.min(P2_x), np.min(P3_x))
+xmax = max(np.max(P1_x), np.max(P2_x), np.max(P3_x))
+ymin = min(np.min(P1_y), np.min(P2_y), np.min(P3_y))
+ymax = max(np.max(P1_y), np.max(P2_y), np.max(P3_y))
+
+width, height = par_map[lx]/3., par_map[lx]/3.
+
+def animate_pendulum(time, P1_x, P1_y, P2_x, P2_y):
+    
+    fig, ax = plt.subplots(figsize=(8, 8), subplot_kw={'aspect': 'equal'})
+    
+    ax.axis('on')
+    ax.set_xlim(xmin - 1., xmax + 1.)
+    ax.set_ylim(ymin - 1., ymax + 1.)
+    
+    ax.set_xlabel('X direction', fontsize=15)
+    ax.set_ylabel('Y direction', fontsize=15)
+    ax.axhline(0, color='black', lw=2)
+
+    line1, = ax.plot([], [], 'o-', lw=0.5, color='blue')    
+    line2, = ax.plot([], [], 'o-', lw=0.5, color='green')                    
+    
+    recht = patches.Rectangle((P1_x[0] - width/2, P1_y[0] - height/2), 
+            width=width, height=height, fill=True, color='red', ec='black')
+    ax.add_patch(recht)
+    return fig, ax, line1, line2, recht
+
+duration = (num_nodes - 1) * solution[-1]
+times = np.linspace(0.0, duration, num=num_nodes)
+fig, ax, line1, line2, recht = animate_pendulum(times, P1_x, P1_y, P2_x, P2_y)
+
+def animate(i):
+    message = (f'running time {times[i]:.2f} sec')
+    ax.set_title(message, fontsize=15)
+    recht.set_xy((P1_x[i] - width/2., P1_y[i] - height/2.))
+
+    wert_x = [P1_x[i], P2_x[i]]
+    wert_y = [P1_y[i], P2_y[i]]               
+    line1.set_data(wert_x, wert_y)    
+
+    wert_x = [P2_x[i], P3_x[i]]
+    wert_y = [P2_y[i], P3_y[i]]
+    line2.set_data(wert_x, wert_y)              
+    return line1, line2,
+
+anim = animation.FuncAnimation(fig, animate, frames=num_nodes,
+                                   interval=40,
```

Comment:
```suggestion
                                   interval=solution[-1]*1000,
```
the interval needs to be a function of the found h.